contracts/os/andromeda-kernel/src/ibc.rs

```diff
@@ -243,42 +243,39 @@ pub fn validate_order_and_version(
     }))
 }
 
-// IBC transfer port
-const TRANSFER_PORT: &str = "transfer";
+pub fn get_counterparty_denom(
+    deps: &Deps,
+    denom: &str,
+    src_channel: &str,
+) -> Result<(String, DenomInfo), ContractError> {
+    // if denom is ibc denom, get denom trace
+    let denom_trace = if denom.starts_with("ibc/") {
+        let ibc_registry_addr = KERNEL_ADDRESSES.load(deps.storage, IBC_REGISTRY_KEY)?;
+        AOSQuerier::denom_trace_getter(&deps.querier, &ibc_registry_addr, denom)?
```

Comment:
This will error if we're wrapping an `ibc/...` denom, right? We should probably have it catch errors and return an empty vector